### PR TITLE
CMake: in case of bad "ALLOCATOR" selected issue warning

### PR DESCRIPTION
**Pull request by aclamk**
Signed-off-by: Adam Kupczyk <akupczyk@redhat.com>

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -295,6 +295,9 @@ elseif(ALLOCATOR STREQUAL "jemalloc")
   set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
 elseif(ALLOCATOR STREQUAL "libc")
   set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
+else()
+  set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
+  message(WARNING "Possibly incorrect allocator selected:" ${ALLOCATOR})
```
Comment (rzarzynski) [inline src/CMakeLists.txt]: Can we raise an error here?

Comment (tchaikov) [reply src/CMakeLists.txt]: yes, we should error out here with an error message.

Comment (wjwithagen) [reply src/CMakeLists.txt]: Please don't.
FreeBSD has jemalloc as default allocator. 
Defining it however, will show that the additional tools are not available in ways that are expected.
So for example rocksdb starts doing wrong things.
If you want to error out, then please make an exemption for FreeBSD.

Comment (tchaikov) [reply src/CMakeLists.txt]: it's always set in `$top_srcdir/CMakeLists.txt`.

Comment (wjwithagen) [reply src/CMakeLists.txt]: You are right. So FreeBSD ends in libc.
Why not error out earlier, like in `$top_srcdir/CMakeLists.txt`?

**Comment by tchaikov** [comment]
@aclamk ping?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -295,6 +295,9 @@ elseif(ALLOCATOR STREQUAL "jemalloc")
   set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
 elseif(ALLOCATOR STREQUAL "libc")
   set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
+else()
+  set(TCMALLOC_srcs perfglue/disabled_heap_profiler.cc)
+  message(FATAL_ERROR "Possibly incorrect allocator selected:" ${ALLOCATOR})
```
Comment (tchaikov) [inline src/CMakeLists.txt]: nit, add a space after `:`. BTW, could put the variable right into the string to print, also could use a stronger tone here. like
```cmake
message(FATAL_ERROR "Unsupported allocator selected: ${ALLOCATOR}")
```

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2017-09-21_06:22:45-rados-wip-kefu-testing-2017-09-21-1013-distro-basic-mira/

**Pull request merged**